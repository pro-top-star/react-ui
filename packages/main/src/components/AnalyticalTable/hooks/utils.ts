export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
