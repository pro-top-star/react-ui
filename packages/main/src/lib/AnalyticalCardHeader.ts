import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
