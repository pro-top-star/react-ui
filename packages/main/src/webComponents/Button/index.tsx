import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Button from '@ui5/webcomponents/dist/Button';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface ButtonPropTypes extends WithWebComponentPropTypes {
  design?: ButtonDesign; // @generated
  disabled?: boolean; // @generated
  icon?: string; // @generated
  iconEnd?: boolean; // @generated
  submits?: boolean; // @generated
  onClick?: (event: Event) => void; // @generated
  children?: string; // @generated
}

/**
 * <code>import { Button } from '@ui5/webcomponents-react/lib/Button';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Button" target="_blank">UI5 Web Components Playground</a>
 */
const Button: FC<ButtonPropTypes> = withWebComponent<ButtonPropTypes>(UI5Button);

Button.displayName = 'Button';

Button.defaultProps = {
  design: ButtonDesign.Default // @generated
};

export { Button };
