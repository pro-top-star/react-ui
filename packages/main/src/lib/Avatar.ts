import { Avatar } from '../webComponents/Avatar';

export { Avatar };
