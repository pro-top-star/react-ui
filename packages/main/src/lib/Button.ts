import { Button } from '../webComponents/Button';

export { Button };
