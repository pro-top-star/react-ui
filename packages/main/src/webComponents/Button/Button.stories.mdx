import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import '@ui5/webcomponents-icons/dist/employee.js';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="Inputs / Button"
  component={Button}
  argTypes={{
    ...createSelectArgTypes({ design: ButtonDesign }),
    children: { control: 'text' },
    ...DocsCommonProps
  }}
  args={{
    design: ButtonDesign.Default,
    icon: 'employee',
    children: 'Button Text',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Button {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

## Usage

For the `Button` UI, you can define text, icon, or both. You can also specify whether the text or the icon is displayed first.  
  
You can choose from a set of predefined types that offer different styling to correspond to the triggered action.  
  
You can set the `Button` as enabled or disabled. An enabled `Button` can be pressed by clicking or tapping it. The button changes its style to provide visual feedback to the user that it is pressed or hovered over with the mouse cursor. A disabled `Button` appears inactive and cannot be pressed.
