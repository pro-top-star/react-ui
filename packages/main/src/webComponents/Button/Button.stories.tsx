import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import React from 'react';

export default {
  title: 'UI5 Web Components / Button',
  component: Button,
  argTypes: {
    ...createSelectArgTypes({ design: ButtonDesign }),
    ref: {
      type: null
    }
  },
  args: {
    design: ButtonDesign.Default,
    icon: 'add',
    children: 'Button Text'
  }
};

export const generatedDefaultStory = ({ design, disabled, icon, iconEnd, submits, onClick, children }) => {
  return (
    <Button design={design} disabled={disabled} icon={icon} iconEnd={iconEnd} submits={submits} onClick={onClick}>
      {children}
    </Button>
  );
};

generatedDefaultStory.storyName = 'Generated default story';
