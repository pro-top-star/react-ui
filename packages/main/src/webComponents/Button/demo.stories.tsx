import { action } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import React from 'react';

const customStyle1 = {
  color: 'red'
};

const customStyle2 = {
  width: '15rem'
};

class DemoButton extends React.Component {
  state = {
    showCS1: false,
    showCS2: false
  };

  render() {
    const { showCS1, showCS2 } = this.state;

    const innerStyles = {
      ...(showCS1 ? customStyle1 : {}),
      ...(showCS2 ? customStyle2 : {})
    };

    return (
      <div>
        <div>
          <Button onClick={() => this.setState({ showCS1: !showCS1 })}>Toggle Custom Style 1</Button>
          <Button onClick={() => this.setState({ showCS2: !showCS2 })}>Toggle Custom Style 2</Button>
        </div>
        <Button
          design={select('design', ButtonDesign, ButtonDesign.Default)}
          disabled={boolean('disabled', false)}
          icon={'add'}
          iconEnd={boolean('iconEnd', false)}
          onClick={action('Button clicked')}
          innerStyles={innerStyles}
        >
          Some Content
        </Button>
      </div>
    );
  }
}

export default {
  title: 'UI5 Web Components / Button',
  component: Button
};

export const generatedDefaultStory = () => (
  <Button
    design={select('design', ButtonDesign, ButtonDesign.Default)}
    disabled={boolean('disabled', false)}
    icon={'add'}
    iconEnd={boolean('iconEnd', false)}
    onClick={action('onClick')}
  >
    Some Content
  </Button>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};

export const withCustomStyles = () => <DemoButton />;

withCustomStyles.story = {
  name: 'With custom Styles'
};
