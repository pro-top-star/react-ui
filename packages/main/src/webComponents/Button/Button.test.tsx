import { mountThemedComponent } from '@shared/tests/utils';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import React from 'react';

describe('Button', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Button />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
