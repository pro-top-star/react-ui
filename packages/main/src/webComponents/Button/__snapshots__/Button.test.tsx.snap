// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-button
    design="Default"
  />
</DocumentFragment>
`;
